### PR TITLE
Add coverage for `np.argwhere`

### PR DESCRIPTION
**Pull request by EPronovost**
Add coverage for `np.argwhere`

### CODE REVIEW

**Comment by esc** [comment]
@EPronovost thanks for submitting this! We appreciate your efforts to improve Numba. I've labelled this as 'in progress' for the time being. Do let us know when it is ready for a review and we will queue it.

**Comment by EPronovost** [comment]
@esc I think this is ready for review.  It's passing all builds except for the Travis CI.  I'm not where to find the logs from that.

File: numba/targets/arraymath.py [hunk 1]
```diff
@@ -2520,6 +2520,15 @@ def np_corrcoef_impl_single_variable(x, y=None, rowvar=True):
 #----------------------------------------------------------------------------
 # Element-wise computations
 
+@overload(np.argwhere)
+def np_argwhere(a):
+
+    def impl(a):
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: I'm not sure that Numba will support all types for the argument `a`, what happens if `a` is a string?

This guide should help describe the sort of checks that should take place http://numba.pydata.org/numba-doc/latest/extending/overloading-guide.html

File: numba/targets/arraymath.py [hunk 2]
```diff
@@ -2520,6 +2520,15 @@ def np_corrcoef_impl_single_variable(x, y=None, rowvar=True):
...
+        return np.transpose(np.vstack(np.nonzero(a)))
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: I think `a` could have `np.asarray()` called on it first, tuples, scalars etc ought to just work.
```pycon
In [15]: np.argwhere((1, 2, 3))                                                                                                  
Out[15]: 
array([[0],
       [1],
       [2]])

In [16]: np.argwhere(1)                                                                                                         
Out[16]: array([[0]])

```

**Comment by esc** [comment]
This is waiting on #4751 right?

**Comment by EPronovost** [comment]
Yes I think that PR should fix the tests failing here.  After advice from @stuartarchibald I believe that branch should pass all tests now, but evidently forgot to push my commit on Monday (just did).  How should I integrate those changes into this PR?  I think I remember being told that merging in from `master` was preferred.

**Comment by esc** [comment]
@EPronovost thanks for pushing now. Yes, the best way to do this is to wait for #4715 to be merged to master and then merge master into this PR.

File: numba/targets/arraymath.py [hunk 3]
```diff
@@ -2535,10 +2552,10 @@ def impl(a):
             return np.nonzero(np.ravel(arr))[0]
     else:
         def impl(a):
-            if a is None:
-                data = [x for x in range(0)]
-            else:
+            if a is not None and bool(a):
                 data = [0]
+            else:
+                data = [x for x in range(0)]
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: I think this change is a bit out of scope for this PR (not sure `argwhere` needs it?) but I think it can be accepted if appropriate unit tests are added.

File: numba/targets/arraymath.py [hunk 4]
```diff
@@ -2520,6 +2520,23 @@ def np_corrcoef_impl_single_variable(x, y=None, rowvar=True):
...
+@overload(np.argwhere)
+def np_argwhere(a):
+
+    if type_can_asarray(a):
+        def impl(a):
+            arr = np.asarray(a)
+            return np.transpose(np.vstack(np.nonzero(arr)))
+    else:
+        def impl(a):
+            if a is not None and bool(a):
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: This won't work with `OptionalType`s but it's out of scope to fix that.

File: numba/targets/arraymath.py (hunk 3 above)
Comment (esc) [reply numba/targets/arraymath.py]: I removed the change and pushed.

**Comment by esc** [comment]
The last minor thing has been resolved and now waiting on CI
